tests/h/services/group_scope_test.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import pytest
+import mock
+
+from h.services.group_scope import group_scope_factory, GroupScopeService
+
+
+class TestFetchByOrigin(object):
+    def test_it_proxies_to_scope_util_for_origin_parsing(
+        self, svc, scope_util, document_uri
+    ):
+        scope_util.parse_origin.return_value = "parsed"
+        svc.fetch_by_origin(document_uri)
+
+        scope_util.parse_origin.assert_called_once_with(document_uri)
+
+    def test_it_returns_empty_list_if_origin_not_parseable(
+        self, svc, scope_util, document_uri
+    ):
+        scope_util.parse_origin.return_value = None
+        scopes = svc.fetch_by_origin(document_uri)
+
+        assert scopes == []
+
+    @pytest.mark.parametrize(
+        "uri,should_match",
+        [
+            ("https://www.foo.com", False),
+            ("http://foo.com", True),
+            ("https://foo.com/bar", False),
+            ("http://foo.com/bar/baz/", True),
+            ("http://www.foo.com/bar/baz.html", False),
+            ("randoscheme://foo.com", False),
+            ("foo", False),
+            ("foo.com", False),
+            ("http://foo.com/bar/baz.html?query=whatever", True),
+            ("", False),
+            (None, False),
+        ],
+    )
+    def test_it_returns_scopes_that_match_uri_origin(
+        self, svc, sample_scopes, uri, should_match
+    ):
+        matches = svc.fetch_by_origin(uri)
+
+        if should_match:
+            assert matches == sample_scopes
+        else:
+            assert matches == []
+
+
+class TestFetchByScope(object):
+    def test_it_proxies_to_fetch_by_origin(self, svc, document_uri):
+        svc.fetch_by_origin = mock.Mock(return_value=[])
+
+        svc.fetch_by_scope(document_uri)
+
+        svc.fetch_by_origin.assert_called_once_with(document_uri)
```

Comment:
I think this is testing an internal implementation detail, and with a dodgy mock